### PR TITLE
add default database for alter table

### PR DESCRIPTION
**Pull request by anneji-dev**
I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en

Changelog category (leave one):
- Improvement

Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
add default database setting for alter table ... on cluster ... replace/move partition from/to ...

Detailed description / Documentation draft:
The test cases are as follows:
```
create database test_replace_partition_move_partition on cluster test_cluster；
use test_replace_partition_move_partition;
create table t1_all on cluster test_cluster(partition_col_1 String, tc1 int,tc2 int)engine=Distributed(test_cluster,test_replace_partition_move_partition,t1_local);
create table t1_local on cluster test_cluster(partition_col_1 String, tc1 int,tc2 int)engine=MergeTree() partition by partition_col_1 order by tc1;

create table t2_all on cluster test_cluster(partition_col_1 String, tc1 int,tc2 int)engine=Distributed(test_cluster,test_replace_partition_move_partition,t2_local);
create table t2_local on cluster test_cluster(partition_col_1 String, tc1 int,tc2 int)engine=MergeTree() partition by partition_col_1 order by tc1;

insert into t1_all values('partition1', 1,1);
insert into t1_all values('partition2', 1,1);
insert into t2_all values('partition1', 3,3);
insert into t2_all values('partition2', 6,6);

ALTER TABLE t1_local on cluster test_cluster REPLACE PARTITION 'partition1' FROM  t2_local；
error:Received from 127.0.0.1:9001. DB::Exception: There was an error on [9.55.181.37:9001]: Code: 60, e.displayText() = DB::Exception: Table default_1.t2_local doesn't exist (version 21.7.1.1).

```
so, we should be add default database for alter table

### CODE REVIEW

**Comment by evillique** [comment]
It would be great to add a test for this :)

File: tests/queries/0_stateless/02028_add_default_database_for_alterquery_on_cluster.sql [hunk 1]
```diff
@@ -0,0 +1,17 @@
+-- Tags: distributed, no-parallely
```
Comment (evillique) [inline tests/queries/0_stateless/02028_add_default_database_for_alterquery_on_cluster.sql:1]:
```suggestion
-- Tags: distributed, no-parallel
```

**Comment by anneji-dev** [comment]
@evillique  Functional stateless tests why failed??? 02028_add_default_database_for_alterquery_on_cluster.reference should be 
0 rows??  All checks must be successful ??

File: tests/queries/0_stateless/02028_add_default_database_for_alterquery_on_cluster.sql [hunk 2]
```diff
@@ -0,0 +1,17 @@
+-- Tags: distributed, no-parallel
```
Comment (evillique) [inline tests/queries/0_stateless/02028_add_default_database_for_alterquery_on_cluster.sql:1]: Seems like it should fix the problem.
```suggestion
-- Tags: distributed, no-parallel, no-replicated-database
-- Tag no-replicated-database: ON CLUSTER is not allowed

```

**Comment by anneji-dev** [comment]
@evillique  I don't understand why failed?  I should use with replicas test clusters?

**Comment by anneji-dev** [comment]
@evillique   I don't understand why failed? Please help me，thanks!

**Comment by evillique** [comment]
I think it fails for the reasons unrelated to your changes. So you only need to sign the CLA and I will merge this PR :)

**Comment by anneji-dev** [comment]
@evillique  ok,thanks very mush

**Comment by anneji-dev** [comment]
@evillique  I have been signed the CLA, please merge this PR:) thanks very much!

**Comment by evillique** [comment]
It seems like CLAassistant treats **anneji** as a separate committer and won't let me merge until you add email that you used for that commit to your **anneji-dev** GitHub profile as explained in https://github.com/ClickHouse/ClickHouse/pull/28955#issuecomment-928975205.

**Pull request merged**